ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -66,48 +156,89 @@ public BlobId(String id, ClusterMap clusterMap) throws IOException {
    * @param clusterMap of the cluster that the blob id belongs to
    * @throws IOException
    */
-  public BlobId(DataInputStream stream, ClusterMap clusterMap) throws IOException {
-    this(stream, clusterMap, false);
+  public static BlobId fromDataInputStream(DataInputStream stream, ClusterMap clusterMap) throws IOException {
```

Comment:
On the other day I had a discussion with @nsivabalan . The way we are going to adopt is: 1) using `builder`s to build objects that take arguments; and 2) using static methods to deserialize a stream/json/file etc. into an object. If these are the only two ways to get an object, then the constructors should be made at least package-private.